File: Problems_and_Solutions/days-10 to 19/day-13/Java/Solution3.java
class Node {
    int data;
    Node next, prev;
    Node(int data) {
        this.data = data;
    }
}

class Solution3 {
    public static Node constructDLL(int[] arr) {
        if (arr.length == 0) return null;

        Node head = new Node(arr[0]);
        Node curr = head;

        for (int i = 1; i < arr.length; i++) {
            Node newNode = new Node(arr[i]);
            curr.next = newNode;
            newNode.prev = curr;
            curr = newNode;
        }

        return head;
    }

    public static void print(Node head) {
        Node curr = head;
        while (curr != null) {
            System.out.print(curr.data);
            if (curr.next != null) {
                System.out.print(" <-> ");
            }
            curr = curr.next;
        }
    }

    public static void main(String[] args) {
        int[] arr = {1, 2, 3, 4, 5};
        Node head = constructDLL(arr);

        print(head);
    }
}

File: Problems_and_Solutions/days-10 to 19/day-13/c++/solution 3.cpp

class Solution {
  public:
    Node* constructDLL(vector<int>& arr) {
        // code here
        Node* head=new Node(arr[0]);
        Node* trav=head;
        for(int i=1;i<arr.size();i++)
        {
            trav->next=new Node(arr[i]);
            trav->next->prev=trav;
            trav=trav->next;
        }
        return head;
    }
};
